### PR TITLE
MQE: Add support for clamp_min and clamp_max

### PR DESCRIPTION
**Pull request by jhesketh**
#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by jhesketh** [comment]
Putting into draft until https://github.com/grafana/mimir/pull/9642 is merged (this will need a similar fix)

File: pkg/streamingpromql/testdata/ours/functions.test
[the remark this answers is not in the document]
Comment (jhesketh) [reply pkg/streamingpromql/testdata/ours/functions.test:1]: Yes, I need to enable them still.

File: pkg/streamingpromql/functions/math.go [hunk 1]
```diff
@@ -86,3 +86,26 @@ var Clamp InstantVectorSeriesFunction = func(seriesData types.InstantVectorSerie
...
+	}
+	if isMin {
+		clampFunc = func(val, f float64) float64 {
+			return max(val, f)
+		}
+	}
+	return func(seriesData types.InstantVectorSeriesData, scalarArgsData []types.ScalarData, memoryConsumptionTracker *limiting.MemoryConsumptionTracker) (types.InstantVectorSeriesData, error) {
+		clampTo := scalarArgsData[0]
+		for step, data := range seriesData.Floats {
+			val := clampTo.Samples[step].F
```
[the remark this answers is not in the document]
Comment (jhesketh) [reply pkg/streamingpromql/functions/math.go:102]: Yep, that's why this is put back into draft. #9642 comes with a test that I'll expand for these functions.

File: pkg/streamingpromql/functions/math.go [hunk 2]
```diff
@@ -86,3 +86,26 @@ var Clamp InstantVectorSeriesFunction = func(seriesData types.InstantVectorSerie
 	seriesData.Histograms = nil
 	return seriesData, nil
 }
+
+func ClampMinMaxFactory(isMin bool) InstantVectorSeriesFunction {
+	clampFunc := func(val, f float64) float64 {
+		return min(val, f)
+	}
```
[the remark this answers is not in the document]
Comment (jhesketh) [reply pkg/streamingpromql/functions/math.go:93]: This doesn't work because they are built-ins: `min (built-in) must be called)`

**Review by charleskorn — APPROVED** [review]
🚀

**Pull request merged**